models/ed/R/sample.IC.ED2.R

```diff
@@ -0,0 +1,114 @@
+##' Samples intial conditions for ED2
+##' 
+##' NOTE: almost all of the fcn args can be extracted from pecan settings list, just wnated to be explicit
+##' 
+##' @param samples a dataframe with columns: tree id, spp info, Measurement_Year, DBH, plot id (if applicable), could come from load_data
+##' @param format_name how the spp info provided: usda, fia, latin_name, custom
+##' @param start_date settings$run$start.date
+##' @param pfts settings$pfts
+##' @param ne number of ensembles requested
+##' @param latitude settings$run$site$lat
+##' @param longitude settings$run$site$lon
+##' @param source the source of the samples, will appear as prefix in the IC filenames e.g. "HF_lyford.PalEON"
+##' @param metadata any known metadata: age, area (to calculate density) etc.
+##' @param outfolder settings$rundir
+##' @param host_info host info, e.g. settings$host
+##' @param inputs_path dir path to inputs, e.g. dirname(settings$run$inputs$css$path)
+##' 
+##' @return IC 
+##' 
+##' @author Istem Fer
+##' @export
+##' 
+sample.IC.ED2 <- function(samples, format_name = "usda", start_date, pfts, ne, 
+                          latitude, longitude, source, metadata, outfolder, host_info, inputs_path) {
+  
+
+  
+  year <- lubridate::year(start_date)-1 # year before the initial run
+  
+  # subset samples for the year, if coming from load_data won't need this
+  samples <- samples[samples$Measurement_Year == year, ]
+  
+  # remove rows with NAs (we don't want DBH to be NA but do we want to allow missing taxa?)
+  samples <- samples[complete.cases(samples), ]
+
+  # match spp info with species names 
+  # the matching can go into the ic-loop below, depending which is easier to handle sample size vs ensemble size
+  # for sample sizes of 10^5 this is still fast without sampling
+  spp_match <- match_species_id(input_codes = samples[[format_name]], format_name = format_name)
+  
+  # match pfts
+  pft_match   <- match_pft(bety_species_id = spp_match$bety_species_id, pfts = pfts) 
+  samples$pft <- pft_match$pft
+  
+  # how many pathces we want? should "n.patches" be a function argument?
+  if(!is.null(samples$plot)){
+    n.patches <- length(unique(samples$plot))
+  }else{
+    n.patches <- 1
+  }
+  
+  ic.list <- list()
+  for(icn in seq_len(ne)){
+    
+    veg_info <- list()
+    
+    # prepare info for css/pss/site
+    css_info    <- list()
+    pss_info    <- list()
+    # site_info <- list()
+    
+    # css
+    patch.list <- list()
+    for(np in seq_len(n.patches)){
+      patch_sub  <- samples[samples$plot == np,] # these colnames might need generalization
+      samp_ind   <- tapply(seq_along(patch_sub$tree), patch_sub$tree, sample, 1)
```

Comment:
Not 100% sure I follow what this line is doing. Could use a comment and a bit of an explanation about the underlying assumptions.